coretex/entities/dataset/network_dataset.py

```diff
@@ -25,12 +25,21 @@
 import logging
 
 from .dataset import Dataset
+from .state import DatasetState
 from ..sample import NetworkSample
 from ... import folder_manager
 from ...codable import KeyDescriptor
-from ...networking import NetworkObject
+from ...networking import NetworkObject, networkManager
 from ...threading import MultithreadedDataProcessor
 
+class EntityNotCreated(Exception):
+
+    """
+        Exception which is raised due to any unexpected behaviour with dataset creation
```

Comment:
Move this exception to `NetworkObject` file and update description. We will start using it there at one point.